index.js

```diff
@@ -88,27 +104,54 @@ class CSVPlugin {
     return this.context.json.expand(data)
   }
 
+  loadExportTemplates(data) {
+    const state = this.context.window.store?.getState()
+    const inferredItemTemplate = data['@graph']?.[0]?.template
+    const itemTemplate = loadTemplate(
+      state,
+      this.options.itemTemplate,
+      inferredItemTemplate)
+
+    if (!itemTemplate)
+      // eslint-disable-next-line max-len
+      throw new Error(`Failed to find specified item template "${this.options.itemTemplate}" or fallback "${data['@graph']?.[0]?.template}". Please install one of these templates, or select a different template and try again.`)
+
+
+    const inferredPhotoTemplate = data['@graph']?.[0]?.photo?.[0]?.template
+    const photoTemplate = this.options.photoMetadata ? loadTemplate(
+      state,
+      this.options.photoTemplate,
+      inferredPhotoTemplate) : null
+
+    if (this.options.photoMetadata && !photoTemplate)
+      // eslint-disable-next-line max-len
+      throw new Error(`Failed to find specified photo template "${this.options.photoTemplate}" or fallback "${data['@graph']?.[0]?.photo?.[0]?.template}". Please install one of these templates, or select a different template and try again.`)
+
+    return { itemTemplate, photoTemplate }
+  }
+
+  maxPhotos(data) {
+    return (this.options.photoMetadata || this.options.photoNotes) ?
+      Math.max(...data['@graph'].map(item => item.photo?.length || 0)) : 0
```

Comment:
Oh, that's a cool style! I think I've always used `.reduce` with `Math.max` inside the reducer, but this reads very well.